test/integration/targets/hcloud_network/tasks/main.yml

```diff
@@ -0,0 +1,91 @@
+# Copyright: (c) 2019, Hetzner Cloud GmbH <info@hetzner-cloud.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+---
+- name: test missing ip_range parameter on create Network
+  hcloud_network:
+    name: "{{hcloud_network_name}}"
+  register: result
+  ignore_errors: yes
+- name: verify fail missing ip_range parameter on create Network result
+  assert:
+    that:
+      - result is failed
+      - 'result.msg == "missing required arguments: ip_range"'
+
+- name: test create Network with check mode
+  hcloud_network:
+    name: "{{hcloud_network_name}}"
+    ip_range: "10.0.0.0/8"
+  register: result
+  check_mode: yes
+- name: verify create Network with check mode result
+  assert:
+    that:
+      - result is changed
+
+- name: test create Network
+  hcloud_network:
+    name: "{{hcloud_network_name}}"
+    ip_range: "10.0.0.0/8"
+  register: network
+- name: verify test create Network result
+  assert:
+    that:
+      - network is changed
+      - network.hcloud_network.name == "{{hcloud_network_name}}"
+      - network.hcloud_network.ip_range == "10.0.0.0/8"
+
+- name: test create Network idempotence
+  hcloud_network:
+    name: "{{hcloud_network_name}}"
+    ip_range: "10.0.0.0/8"
+  register: network
+- name: verify test create Network idempotence result
+  assert:
+    that:
+      - network is not changed
+      - network.hcloud_network.name == "{{hcloud_network_name}}"
+      - network.hcloud_network.ip_range == "10.0.0.0/8"
+
+- name: test create Network idempotence
```

Comment:
Why do you check this another time? Or did you wanted to do this test with check_mode?

Reply:
@resmo told us that we should have an explicit idempotency check.

Reply:
But you already have one in the test above, so there's no need to do a second idempotency check?

Reply:
whoops, copy and paste error... sorry!